spark-rapids/spark-rapids.sh

```diff
@@ -0,0 +1,562 @@
+  systemctl start dataproc-cgroup-device-permissions
+}
+
+function setup_gpu_yarn() {
+  if [[ ${OS_NAME} != debian ]] && [[ ${OS_NAME} != ubuntu ]] && [[ ${OS_NAME} != rocky ]]; then
+    echo "Unsupported OS: '${OS_NAME}'"
+    exit 1
+  fi
+
+  if [[ ${OS_NAME} == debian ]] || [[ ${OS_NAME} == ubuntu ]]; then
+    export DEBIAN_FRONTEND=noninteractive
+    execute_with_retries "apt-get update"
+    execute_with_retries "apt-get install -y -q pciutils"
+  elif [[ ${OS_NAME} == rocky ]] ; then
+    execute_with_retries "dnf -y -q update"
+    execute_with_retries "dnf -y -q install pciutils"
+    execute_with_retries "dnf -y -q install kernel-devel"
+    execute_with_retries "dnf -y -q install gcc"
+  fi
+
+  # This configuration should be ran on all nodes
+  # regardless if they have attached GPUs
+  configure_yarn
+
+  # Detect NVIDIA GPU
+  if (lspci | grep -q NVIDIA); then
+    # if this is called without the MIG script then the drivers are not installed
+    if (/usr/bin/nvidia-smi --query-gpu=mig.mode.current --format=csv,noheader | uniq | wc -l); then
+      NUM_MIG_GPUS=`/usr/bin/nvidia-smi --query-gpu=mig.mode.current --format=csv,noheader | uniq | wc -l`
+      if [[ $NUM_MIG_GPUS -eq 1 ]]; then
+        if (/usr/bin/nvidia-smi --query-gpu=mig.mode.current --format=csv,noheader | grep Enabled); then
+          IS_MIG_ENABLED=1
+          NVIDIA_SMI_PATH='/usr/local/yarn-mig-scripts/'
+          MIG_MAJOR_CAPS=`grep nvidia-caps /proc/devices | cut -d ' ' -f 1`
+          fetch_mig_scripts
+        fi
+      fi
+    fi
+
+    if [[ ${OS_NAME} == debian ]] || [[ ${OS_NAME} == ubuntu ]]; then
+      execute_with_retries "apt-get install -y -q 'linux-headers-$(uname -r)'"
+    fi
+
+    # if mig is enabled drivers would have already been installed
+    if [[ $IS_MIG_ENABLED -eq 0 ]]; then
+      install_nvidia_gpu_driver
+      if [[ -n ${CUDNN_VERSION} ]]; then
+        install_nvidia_nccl
+        install_nvidia_cudnn
+      fi
+      #Install GPU metrics collection in Stackdriver if needed
+      if [[ ${INSTALL_GPU_AGENT} == true ]]; then
+        install_gpu_agent
+        echo 'GPU metrics agent successfully deployed.'
+      else
+        echo 'GPU metrics agent will not be installed.'
+      fi
+      configure_gpu_exclusive_mode
+    fi
+
+    configure_yarn_nodemanager
+    configure_gpu_script
+    configure_gpu_isolation
+  elif [[ "${ROLE}" == "Master" ]]; then
+    configure_yarn_nodemanager
+    configure_gpu_script
+  fi
+
+  # Restart YARN services if they are running already
+  if [[ $(systemctl show hadoop-yarn-resourcemanager.service -p SubState --value) == 'running' ]]; then
+    systemctl restart hadoop-yarn-resourcemanager.service
+  fi
+  if [[ $(systemctl show hadoop-yarn-nodemanager.service -p SubState --value) == 'running' ]]; then
+    systemctl restart hadoop-yarn-nodemanager.service
+  fi
+}
+function main() {
+  setup_gpu_yarn
+  
+  if [[ "${RUNTIME}" == "SPARK" ]]; then
```

Comment:
Can we remove the need for specifying the RUNTIME metadata variable for spark-rapids.sh since the init script is specific to Spark?

Reply:
@medb what is your thought on this? I feel maybe this can be a safe check just in case people are using this script in a wrong "RUNTIME".